array_api_tests/hypothesis_helpers.py

```diff
@@ -121,8 +125,17 @@ def two_broadcastable_shapes(draw, shapes=shapes):
 
 nonbroadcastable_ones_array_two_args = hypotheses_tuples(ones_arrays, ones_arrays)
 
-# TODO: Generate general arrays here, rather than just scalars.
-numeric_arrays = builds(full, just((1,)), floats())
+floating_arrays = xps.arrays(dtype=xps.floating_dtypes(), shape=xps.array_shapes())
+
+@composite
+def broadcastable_floating_array_pairs(draw):
```

Comment:
I think we instead want a function `broadcastable_array_pairs` (or `mutually_broadcastable_array_pairs`), which takes a dtype strategy as input. This will be similar to the `array_scalars` function in this file, which I expect this strategy will replace most instances of. The specific dtype combinations can then be set as `floating_arrays = broadcastable_array_pairs(floating_dtypes)` and so on. See the stuff at the top of test_elementwise.py (which should probably be moved to this file).